File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.java
class Solution {
    public int solution(String[] babbling) {
        int answer = 0;
        String[] strs = {"aya", "ye", "woo", "ma"};
        String[] str = {"ayaaya", "yeye", "woowoo", "mama"};
        for(int i = 0; i < babbling.length; i++) {
            for(int j = 0; j < str.length; j++) {
                if(!babbling[i].contains(str[j])) {
                    babbling[i] = babbling[i].replace(strs[j], " ");
                }
                
            }
            if(babbling[i].trim().length() == 0) {
                answer++;
            }
            
        }
        return answer;
    }
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.js
function solution(babbling) {
    let answer = 0;
    let speak = ["aya","ye","woo","ma"];
    
    // babbling[i] 를 bab으로 정의
    for(let i = 0; i < babbling.length; i++){
        let bab = babbling[i];
        
        // 두번 연속 발음할 경우 반복문에서 탈출
        for(let j = 0; j < speak.length; j++){
            if(bab.includes(speak[j].repeat(2))){
                break;
            }
            
            // split을 이용해, 말할수 있는 단어를 제거
            bab = bab.split(speak[j]).join(" ");  
        } 
    
            // bab의 문자 갯수가 0 이라면 발음 가능한 단어이므로 answer에 1을 더해준다
            if(bab.split(" ").join("").length === 0){
                answer++;
            }
    }
    return answer;
}